src/main/java/org/apereo/openequella/adminconsole/launcher/ClientLauncher.java

```diff
@@ -0,0 +1,391 @@
+  private static final String CLONE_SERVER_TITLE = "Clone server?";
+  private static final String PROMPT_CLONE = "Do you want to clone the currently selected server?";
+  private static final String REMOVE_SERVER_TITLE = "Remove server?";
+  private static final String PROMPT_REMOVE = "Are you sure you want to remove this server?";
+  private static final String MAKE_DEFAULT_TITLE = "Make default?";
+  private static final String PROMPT_MAKE_DEFAULT = "Make this server the default server to show?";
+
+  private Config config;
+
+  private final ServerPicker serverPicker;
+  private final JButton addServerButton;
+  private final JButton editServerButton;
+  private final JButton removeServerButton;
+  private final JButton makeDefaultServerButton;
+  private final JButton editProxyButton;
+  private final JButton launchButton;
+
+  public static void main(String args[]) throws Exception {
+    new ClientLauncher();
+  }
+
+  public ClientLauncher() {
+    try {
+      UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+      BlindSSLSocketFactory.register();
+
+      config = readConfig();
+      serverPicker = new ServerPicker(config);
+      addServerButton = new JButton(ADD_SERVER_BUTTON);
+      editServerButton = new JButton(EDIT_SERVER_BUTTON);
+      removeServerButton = new JButton(REMOVE_SERVER_BUTTON);
+      editProxyButton = new JButton(PROXY_SETTINGS_BUTTON);
+      makeDefaultServerButton = new JButton(MAKE_DEFAULT_SERVER_BUTTON);
+      launchButton = new JButton(LAUNCH_BUTTON);
+
+      setTitle(WINDOW_TITLE);
+      setResizable(false);
+      addWindowListener(this);
+      setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+      final JComponent generalActions = createGeneralActions();
+      final JComponent serverList = createServerList();
+      final JComponent serverActions = createServerActions();
+
+      final int[] rows = { generalActions.getPreferredSize().height, 0, serverList.getPreferredSize().height, 0,
+          serverActions.getPreferredSize().height };
+      final TableLayout layout = new TableLayout(rows, new int[] { WINDOW_WIDTH });
+      final JPanel all = new JPanel(layout);
+      all.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+      all.add(generalActions, new Rectangle(0, 0, 1, 1));
+      all.add(serverList, new Rectangle(0, 2, 1, 1));
+      all.add(serverActions, new Rectangle(0, 4, 1, 1));
+
+      updateButtons();
+
+      getContentPane().add(all);
+      getRootPane().setDefaultButton(launchButton);
+      addWindowListener(this);
+      pack();
+
+      ComponentHelper.centreOnScreen(this);
+      setVisible(true);
+
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private JComponent createServerList() {
+    final JLabel profileLabel = new JLabel(LABEL_SERVER);
+
+    final JPanel serverPanel = new JPanel(
+        new TableLayout(new int[] { profileLabel.getPreferredSize().height, serverPicker.getPreferredSize().height },
```

Comment:
Maybe here we can have some local variables to help readability.

Reply:
I think that it looks more readable like this.  I'm open to a vote :)